### PR TITLE
fix (cacheable-request): reintroduce temporarily to fix dependency semver

### LINKED ISSUE #62793: Issue with removing typings for previously existing libraries ( specifically, @types/keyv )
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62741 removed typings for @types/keyv library, replacing them with a readme file that suggests not using this package.

This causes errors in compilation for packages that are dependent on @types/keyv -
image

That is to the issue in typescript - https://github.com/microsoft/TypeScript/issues/27956, which basically requires a @types package having index.d.ts to not break compilation.

Removing this library altogether would be good, but not possible due to it being the transitive dependency for my package.

IMO, this change breaks semver and should be fixed as a patch.

### PR DESCRIPTION
**Pull request by 43081j**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [~] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

---

this re-introduces cacheable-request's types **temporarily** for the sole purpose of publishing a version with a specific dependency list.

im guessing if a DT package has no explicit dependencies, it falls back to auto-generating a package.json with a semver range of `*`.

basically `@types/cacheable-request` never had a package manifest, but the published package does, and that specifies `*` for each implicit dependency.

because we want to remove `responselike` and `keyv` from DT (we have for the latter), that `*` range will match the newly published stub package, leading to breaking builds ofc.

so im reintroducing this only so i can specify a range, i'll then follow it up with a PR to remove it again.

@sandersn or another DT maintainer: will this result in the published package dropping back to its original version? (the one specified in the DTS, thats what we want it to do ideally)

depends on microsoft/DefinitelyTyped-tools#561

### CODE REVIEW

**Comment by Ansile** [comment]
@43081j can you please remove "fixes https://github.com/DefinitelyTyped/DefinitelyTyped/issues/62793" in the message? This will close the issue when this pr is merged.

And while the current breakage is fixed, the issue is ultimately about not having same-type breakages, not only here, but for other `@types` packages

**Comment by 43081j** [comment]
didn't realise you had renamed it, have removed it.

**Review by sandersn — APPROVED** [review]
